docs/make_gandiva_func_index.py

```diff
@@ -0,0 +1,128 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+'''Generates a RST file containing all Gandiva function signatures
+
+This is meant as a temporary script until we begin writing docs for Gandiva funcs.'''
```

Comment:
Right now this just lists the available function signatures and does so in an automated way. I hope we later have hand-written documentation for the functions. For example, we likely need to explain somewhere what the difference is between `equal`, `eq`, `same`, and `is_not_distinct_from`. (I myself have no idea 😵.)

I plan on starting a thread in the mailing list to discuss the most maintainable way to setup that function documentation.